src/XIOPAK.CPP

```diff
@@ -650,19 +664,17 @@ LI FC xdisksp()		// returns diskspace on default drive
 //=============================================================================
 LI FC dskDrvFreeSpace()		// returns total disk free space in bytes on current default drive
 {
-	char * rootPath = NULL;	// NULL for current, or "root directory of disk" -- C:\ i guess
-
-	DWORD spc;			// receives sectors per cluster
-	DWORD bps;			// receives bytes per sector
-	DWORD freec;		// receives free clusters
-	DWORD totcl;
+	// char * rootPath = NULL;	// NULL for current, or "root directory of disk" -- C:\ i guess
+	std::error_code spaceErrorCode;
+	filesys::path inputPath = filesys::current_path(); // rootPath - since is defined as NULL I will look for current path
+	filesys::space_info freeSpace = filesys::space(inputPath, spaceErrorCode);
 
-	if ( !GetDiskFreeSpace( rootPath, &spc, &bps, &freec, &totcl))
+	if ( !spaceErrorCode )
 		// GetLastError for more info
 		return 0L;				// call failed, return 0 space avail
 
-	if ( (float)spc * (float)bps * (float)freec < (float)LONG_MAX)	// if free space fits in a long
-		return spc * bps * freec;					// compute fixed point free space & return it
+	if ( freeSpace.available < (float)LONG_MAX)	// if free space fits in a long
+		return freeSpace.available;					// compute fixed point free space & return it
```

Comment:
We talked about testing this, but I didn't see any mention of that in your PR description.

Reply:
How can I add the test file where I prove they both have equal behavior for reference?

Reply:
Good question. CSE isn't yet set up with unit tests (that would be the ideal approach here). I think it's fine just to mention your tests in the PR description. Our conversation here can also help document that you tested these changes and that I reviewed them with you.